File: Selenium/Activities/TestActivity8.java
package activities;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class TestActivity8 {

	public static void main(String[] args) {
WebDriver driver = new FirefoxDriver();
Actions builder = new Actions(driver);
		
		driver.get("https://training-support.net/webelements/mouse-events");
        // Print the title of the page
        System.out.println("Page title: " + driver.getTitle());
        
        
        WebElement lockButton = driver.findElement(By.xpath("//h1[text()='Cargo.lock']"));
        WebElement tomlButton = driver.findElement(By.xpath("//h1[text()='Cargo.toml']"));
        WebElement src = driver.findElement(By.xpath("//h1[text()='src']"));
        WebElement target = driver.findElement(By.xpath("//h1[text()='target']"));
        
        
        
      //left click .lock button and then click .toml button
        builder.click(lockButton).moveToElement(tomlButton).click().build().perform();
        
        //print the message
        System.out.println(driver.findElement(By.id("result")).getText());
        
        //double click src and right click target and select open
        builder.doubleClick(src).contextClick(target).build().perform();
        
        WebElement openOption = driver.findElement(By.xpath("//div[@id='menu']//ul/li[1]"));
        //openOption.click();
        builder.click(openOption).pause(2000L).build().perform();
        
        System.out.println(driver.findElement(By.id("result")).getText());
        driver.quit();

	}

}

File: Selenium/Activities/TestActivity8.py
# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver import ActionChains

# Start the Driver
with webdriver.Firefox() as driver:
    builder = ActionChains(driver)
    # Navigate to the URL
    driver.get("https://training-support.net/webelements/mouse-events")
    # Print the title of the page
    print("Page title is: ", driver.title)

    lock_button = driver.find_element(By.XPATH,"//h1[text()='Cargo.lock']")
    toml_button = driver.find_element(By.XPATH,"//h1[text()='Cargo.toml']")
    src_button = driver.find_element(By.XPATH,"//h1[text()='src']")
    target_button = driver.find_element(By.XPATH,"//h1[text()='target']")

    builder.click(lock_button).move_to_element(toml_button).click().pause(2).perform()
    print(driver.find_element(By.ID,"result").text)

    builder.double_click(src_button).context_click(target_button).perform()

    open_button = driver.find_element(By.XPATH,"//div[@id='menu']//ul/li[1]")

    builder.click(open_button).pause(2).perform()
    print(driver.find_element(By.ID,"result").text)
    driver.quit()


